url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/neworder/addpizza/AddPizzaViewGUI.java

```diff
@@ -0,0 +1,889 @@
+	public Color getTableBGColor(ZScrollTable table) {
+		return new Color( 100, 100, 100);
+	}
+
+	/**
+	 * @see gui.scrolltable.ZScrollTableDelegate#getEmptyTableMessage(gui.scrolltable.ZScrollTable)
+	 */
+	public String getEmptyTableMessage( ZScrollTable table ) {
+		return "There are no toppings on the pizza.";
+	}
+
+	@Override
+	public ZScrollTableCell getCell(ZScrollTable table, int index) {
+		//Create a new ToppingTableCell
+		ToppingTableCell cell = toppingsOptionsCells.get(index);
+		
+		// Color the cell based on the index.
+		int shade = 220 + (index % 2) * 20;
+		cell.setBackground( new Color( shade, shade, shade ) );
+		cell.getToppingsLocationPanel().setBackground(new Color( shade, shade, shade ) );
+		
+		return cell;
+	}
+
+	@Override
+	public int getCellWidth(ZScrollTable table) {
+		// TODO Auto-generated method stub
+		return ToppingTableCell.CELL_WIDTH;
+	}
+
+	@Override
+	public int getCellHeight(ZScrollTable table, int index) {
+		// TODO Auto-generated method stub
+		return ToppingTableCell.CELL_HEIGHT;
+	}
+
+
+
+	@Override
+	public boolean canDeleteCell(ZScrollTable table, int index) {
+		return true;
+	}
+
+
+	@Override
+	public void cellWasDeleted(ZScrollTable table, int index) {
+		// TODO Auto-generated method stub
+			toppingsOptionsCells.remove(index);
+			_toppingsOptionsTable.updateTable();
+			updateProgressBars();
+				
+	}
+
+
+	@Override
+	public JPanel getMainPanel() {
+		// TODO Auto-generated method stub
+		return _mainPanel;
+	}
+
+
+	@Override
+	public void setVisible(boolean visible) {
+		
+		// Pass the change on to the main panel.
+		_mainPanel.setVisible( visible );
+
+		// Show the back button if this is going to be displayed.
+		PDSViewManager.setBackButtonEnabled( visible );
+		
+	}
+
+
+	@Override
+	public void setChannelEnabled(InputChannel inChannel, boolean enabled) {
+		// TODO Auto-generated method stub
+		
+	}
+
+
+	@Override
+	public void displayString(String message, OutputChannel outChannel) {
+		// TODO Auto-generated method stub
+		
+	}
+
+
+	@Override
+	public void displayObject(Object object, OutputChannel outChannel) {
+		// TODO Auto-generated method stub
+		
+	}
+
+
+	@Override
+	public <T> void displayList(ArrayList<T> list, OutputChannel outChannel) {
+		// TODO Auto-generated method stub
+		
+	}
+
+
+	@Override
+	public ZScrollTableCell getHeaderCell(ZScrollTable table) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+
+	@Override
+	public void cellWasClicked(ZScrollTable table, int index) {
+		
+	}
+	
+	public int getNumberOfLeftToppings(){
+		
+		int numToppLeft = 0;
+		for( ToppingTableCell topping : toppingsOptionsCells){
+			String toppingLocationKey = topping.getToppingLocationKey();
+			if( toppingLocationKey.equalsIgnoreCase( AddPizzaView.WHOLE_KEY ) ) {
+				numToppLeft++;
+			} else if( toppingLocationKey.equalsIgnoreCase( AddPizzaView.LEFT_KEY ) ) {
+				numToppLeft++;
+			}
+			else{
+				numToppLeft += 0;
+			}
+		}
+		
+		return numToppLeft;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 819:832, 837:850.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.